### PR TITLE
[ADD] `LSMRInverseLinearOperator`

### LINKED ISSUE #105: [ADD] `LSMRInverseLinearOperator`
Add [lsmr](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.linalg.lsmr.html) to invert linear operators.

### PR DESCRIPTION
**Pull request by runame**
Resolves #105.

Also adds more methods of `KFACLinearOperator` and `KFACInverseLinearOperator` to the docs.

### CODE REVIEW

File: curvlinops/inverse.py [hunk 1]
```diff
@@ -81,6 +80,81 @@ def _matvec(self, x: ndarray) -> ndarray:
...
+
+        # LSMR hyperparameters
+        self.set_lsmr_hyperparameters()
+
+    def set_lsmr_hyperparameters(
+        self,
+        damp=0.0,
+        atol=1e-06,
+        btol=1e-06,
+        conlim=100000000.0,
```
Comment (f-dangel) [inline curvlinops/inverse.py:106]: Can we change this into `1e8` so I don't have to count the zeros every time I want to know the value? 😂

File: curvlinops/inverse.py [hunk 2]
```diff
@@ -81,6 +80,81 @@ def _matvec(self, x: ndarray) -> ndarray:
...
+        maxiter=None,
+        show=False,
+        x0=None,
```
Comment (f-dangel) [inline curvlinops/inverse.py:109]: Add type annotations.

**Review by f-dangel — COMMENTED** [review]
Minor things. Feel free to address and merge.

File: curvlinops/inverse.py (hunk 2 above)
Comment (runame) [reply curvlinops/inverse.py]: Also added type hints for CG.

**Pull request merged**